cmd/serve/main.go

```diff
@@ -1,22 +1,53 @@
 package main
 
 import (
+	"context"
 	"embed"
 	"fmt"
 	"io/fs"
 	"log"
 	"net/http"
+	"os"
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq"
+
+	create_user_v1 "github.com/sunb26/xat/handler/create_user"
 )
 
 //go:embed all:web all:web/_next
 var content embed.FS
 
+type middleware struct {
+	handler http.Handler
+	db      *sqlx.DB
+}
+
+func (i *middleware) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
perhaps move request and response logging to here?

Reply:
Been doing some digging on this. It is trivial to log headers and methods, etc. however it is non-trivial to log the body of both the request and responses in middleware. It seems to involve writing custom implementations for each to read the body and then put it back into the connection stream.

https://github.com/gin-gonic/gin/issues/961
https://stackoverflow.com/questions/66528234/log-http-responsewriter-content

My judgement here is that it would be simpler to read and implement and also about the same number of lines of code to just add a log statement at the start and end of every endpoint to log the req and response. Wdyt?

Reply:
good point, what about logging the req and res headers only?

Reply:
Yeah I can move logging headers to the middleware and print the body content in the handler.